src/main/java/org/kuali/kfs/coa/document/validation/impl/AccountGlobalRule.java

```diff
@@ -863,5 +962,1065 @@ private boolean isIncomeStreamAccountRequired(String fundGroupCode, String subFu
+         }
+    	 
+		// if no active ICR accounts on Global doc then check acct details
+		if (success) {
+			accountGlobalDetail.refreshReferenceObject(KFSPropertyConstants.ACCOUNT);
+			List<IndirectCostRecoveryAccount> activeICRList = new ArrayList<IndirectCostRecoveryAccount>();
+
+			for (IndirectCostRecoveryAccount icrAccount : accountGlobalDetail.getAccount().getActiveIndirectCostRecoveryAccounts()) {
+				boolean foundInactivated = false;
+				for (IndirectCostRecoveryAccountChange icrAccountChange : newAccount.getIndirectCostRecoveryAccounts()) {
+					if (!icrAccountChange.isActive()) {
+						if (icrAccount.getIndirectCostRecoveryFinCoaCode().equalsIgnoreCase(icrAccountChange.getIndirectCostRecoveryFinCoaCode()) && icrAccount.getIndirectCostRecoveryAccountNumber().equalsIgnoreCase(icrAccountChange.getIndirectCostRecoveryAccountNumber())) {
+							foundInactivated = true;
+							break;
+						}
+					}
+				}
+				if (!foundInactivated) {
+					activeICRList.add(icrAccount);
+				}
+			}
+			success &= activeICRList.isEmpty();
+			if (!success) {
+				putFieldError(KFSPropertyConstants.INDIRECT_COST_RECOVERY_ACCOUNTS, CUKFSKeyConstants.ERROR_DOCUMENT_ACCT_GLB_MAINT_ICR_NOT_EMPTY_FOR_NON_CG_ACCOUNT, new String[] { subFundGroupCode, accountGlobalDetail.getAccountNumber() });
+			}
+		}
+    	return success;
+    	
+    }
+    
+    /**
+     * This method checks if the ICR collection should or should not be filled
+     * add error message if validation is not successful
+     * 
+     * @param expectFilled
+     * @param errorMessage
+     * @param args
+     * @return
+     */
+    protected boolean checkICRCollectionExistWithErrorMessage(boolean expectFilled, String errorMessage, String args, CuAccountGlobal newAccount) {
+        boolean success = true;
+        if(expectFilled){
+        	success = !newAccount.getActiveIndirectCostRecoveryAccounts().isEmpty();
+            if(success){
+            		for (IndirectCostRecoveryAccountChange account : newAccount.getActiveIndirectCostRecoveryAccounts()){
+                        success &= StringUtils.isNotBlank(account.getIndirectCostRecoveryAccountNumber())
+                            && StringUtils.isNotBlank(account.getIndirectCostRecoveryFinCoaCode());
+                    }
+            	}
+
+        	if(!success){
+        		if(ObjectUtils.isNotNull(newAccount.getAccountGlobalDetails()) && !newAccount.getAccountGlobalDetails().isEmpty()){
+        			for(AccountGlobalDetail accountGlobalDetail : newAccount.getAccountGlobalDetails()){
+        				accountGlobalDetail.refreshReferenceObject(KFSPropertyConstants.ACCOUNT);
+        				success &= !accountGlobalDetail.getAccount().getActiveIndirectCostRecoveryAccounts().isEmpty();
+        				if(success){
+        					continue;
```

Comment:
This "continue" statement should be removed since it's at the end of the loop anyway.